plugins/aws/aws_vault_importer.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"context"
+	"os"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	"github.com/99designs/aws-vault/v7/cli"
+	"github.com/99designs/aws-vault/v7/vault"
+)
+
+// TryAwsVaultCredentials looks for the access key in the user's vaulting backend through AWS Vault.
+func TryAwsVaultCredentials() sdk.Importer {
+	return TryAWSVault(func(ctx context.Context, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		// Determine the vaulting backend through AWS_VAULT_BACKEND or based on the OS
+		awsVault := &cli.AwsVault{}
+		if awsVaultBackend := os.Getenv("AWS_VAULT_BACKEND"); awsVaultBackend != "" {
+			awsVault.KeyringBackend = awsVaultBackend
+		}
+		keyring, err := awsVault.Keyring()
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+
+		// Use the CredentialKeyring struct from aws-vault to retrieve vaulting backend credentials
+		credentialKeyring := &vault.CredentialKeyring{Keyring: keyring}
+
+		// Load the AWS config file (default location at ~/.aws/config)
+		awsConfigFile, err := awsVault.AwsConfigFile()
```

Comment:
nice that you're using the aws-vault functionality for handling config files!

Reply:
Makes sense, sounds like a good idea to me too. Once the changes for this PR are finalized and merged, I'll create a new issue for the file importer.

By the way, the config file functionality that I used for the AWS Vault importer actually comes from the [AWS Vault SDK](https://pkg.go.dev/github.com/99designs/aws-vault/v7@v7.2.0), not the [AWS SDK](https://aws.amazon.com/sdk-for-go/), in case there was any confusion about that.

Reply:
https://github.com/1Password/shell-plugins/issues/235 Created issue to track these changes.